contracts/ValidateSPV.sol

```diff
@@ -15,187 +15,106 @@ library ValidateSPV {
+            return true;
         }
-    }
-
-    /// @notice         Parses tx outputs
-    /// @param _tx      The raw byte tx
-    /// @return         Raw bytes num outputs and outputs
-    function extractAllOutputs(bytes memory _tx) public pure returns (bytes memory _nOutputs, bytes memory _outputs) {
-        _nOutputs = _tx.extractNumOutputsBytes();
-        uint8 _tmpN = _tx.extractNumOutputs();
 
-        for (uint8 i = 0; i < _tmpN; i++) {
-            _outputs = _outputs.concat(_tx.extractOutputAtIndex(i));
-        }
+        bytes memory _proof = abi.encodePacked(_txid, _intermediateNodes, _merkleRoot);
+        // If the Merkle proof failed, bubble up error
+        return _proof.verifyHash256Merkle(_index);
     }
 
     /// @notice             Hashes transaction to get txid
-    /// @dev                This supports ONLY WITNESS INPUTS AND OUTPUTS
+    /// @dev                This supports legacy now
     /// @param _version     4-bytes version
-    /// @param _nInputs     1-byte num inputs
-    /// @param _inputs      Raw bytes inputs
-    /// @param _nOutputs    1-byte num outputs
-    /// @param _outputs     Raw bytes ouputs
+    /// @param _vin         Raw bytes length-prefixed input vector
+    /// @param _vout        Raw bytes length-prefixed output vector
+    /// @ param _locktime   4-byte tx locktime
     /// @return             32-byte transaction id, little endian
-    function transactionHash(
+    function calculateTxId(
         bytes memory _version,
-        bytes memory _nInputs,
-        bytes memory _inputs,
-        bytes memory _nOutputs,
-        bytes memory _outputs,
+        bytes memory _vin,
+        bytes memory _vout,
         bytes memory _locktime
-    ) public pure returns (bytes32) {
+    ) internal pure returns (bytes32) {
         // Get transaction hash dSha256(version + nIns + inputs + nOuts + outputs + locktime)
-        return abi.encodePacked(_version, _nInputs, _inputs, _nOutputs, _outputs, _locktime).hash256();
-    }
-
-    /// @notice         Validates the first 6 bytes of a transaction
-    /// @dev            First byte is the version. The next must be 0x0000000001
-    /// @param _bytes   Prefixed raw byte string (probably a tx)
-    /// @return         true if valid, otherwise false
-    function validatePrefix(bytes memory _bytes) public pure returns (bool) {
-
-        if (_bytes.length < 6) { return false; }
-
-        bytes32 _versionHash = keccak256(_bytes.slice(0, 1));
-
-        // Return true if prefix is version 1 or 2 and has segwit flag
-        return ((_versionHash == keccak256(hex'01') || _versionHash == keccak256(hex'02'))
-                && keccak256(_bytes.slice(1, 5)) == keccak256(hex'0000000001'));
+        return abi.encodePacked(_version, _vin, _vout, _locktime).hash256();
     }
 
     /// @notice         Parses a tx input from raw input bytes
-    /// @dev            Checks for blank scriptSig
+    /// @dev            Supports Legacy Inputs now too
```

Comment:
We should be explicit that it supports Legacy and Witness inputs. Or maybe we don't need this comment at all as we support all available (2)?